### PR TITLE
Remove deprecated util functions in options_util.h

### PR DESCRIPTION
**Pull request by jowlyzhang**
Remove the util functions in options_util.h that have previously been marked deprecated.

Test Plan:
`make check`

### CODE REVIEW

File: java/rocksjni/options_util.cc [hunk 1]
```diff
@@ -51,40 +51,12 @@ void build_column_family_descriptor_list(
...
-    // error, raise an exception
-    ROCKSDB_NAMESPACE::RocksDBExceptionJni::ThrowNew(env, s);
-  } else {
-    build_column_family_descriptor_list(env, jcfds, cf_descs);
-  }
-}
-
 /*
  * Class:     org_rocksdb_OptionsUtil
  * Method:    loadLatestOptions_1
```
Comment (ltamasi) [inline java/rocksjni/options_util.cc:56]: I suppose we can remove the `_1` suffix now

File: utilities/options/options_util_test.cc [hunk 2]
```diff
@@ -386,11 +377,12 @@ TEST_F(OptionsUtilTest, LatestOptionsNotFound) {
   ASSERT_TRUE(s.IsNotFound());
   ASSERT_TRUE(s.IsPathNotFound());
 
-  s = LoadLatestOptions(dbname_, options.env, &options, &cf_descs);
-  ASSERT_TRUE(s.IsNotFound());
-  ASSERT_TRUE(s.IsPathNotFound());
+  //  s = LoadLatestOptions(dbname_, options.env, &options, &cf_descs);
+  //  ASSERT_TRUE(s.IsNotFound());
+  //  ASSERT_TRUE(s.IsPathNotFound());
```
Comment (ltamasi) [inline utilities/options/options_util_test.cc:382]: This is testing one of the deprecated overloads, right? If so, we can just remove it

File: HISTORY.md [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 ### Public API Changes
 * Completely removed the following deprecated/obsolete statistics: `STALL_L0_SLOWDOWN_MICROS`, `STALL_MEMTABLE_COMPACTION_MICROS`, `STALL_L0_NUM_FILES_MICROS`, `NO_ITERATORS`, `BLOB_DB_GC_NUM_KEYS_OVERWRITTEN`, `BLOB_DB_GC_NUM_KEYS_EXPIRED`, `BLOB_DB_GC_BYTES_OVERWRITTEN`, `BLOB_DB_GC_BYTES_EXPIRED`, `BLOB_DB_GC_MICROS`, and `NUM_DATA_BLOCKS_READ_PER_LEVEL`.
+* Removed the deprecated version of these utility functions: `LoadOptionsFromFile`, `LoadLatestOptions`.
```
[the remark this answers is not in the document]
Comment (ltamasi) [reply HISTORY.md:11]: Also, it might be worth mentioning that the corresponding methods are gone from the Java bindings as well.

**Review by ltamasi — APPROVED** [review]
Thanks for the PR @jowlyzhang ! LGTM

File: HISTORY.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jowlyzhang) [reply HISTORY.md]: Good point. I added these info to the history entry.

File: java/rocksjni/options_util.cc (hunk 1 above)
Comment (jowlyzhang) [reply java/rocksjni/options_util.cc]: Good catch. Thank you for your thorough review!

File: utilities/options/options_util_test.cc (hunk 2 above)
Comment (jowlyzhang) [reply utilities/options/options_util_test.cc]: Yes, thank you for catching this. I forgot to remove it.

**Pull request closed without merging**